pkgs/development/python-modules/pytest-docker/default.nix

```
@@ -0,0 +1,71 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  attrs,
+  docker-compose,
+  mypy,
+  py,
+  pytest,
+  pytestCheckHook,
+  pytest-mypy,
+  pytest-pycodestyle,
+  pytest-pylint,
+  requests,
+  types-requests,
+  types-setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-docker";
+  version = "3.1.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-I3FSSASnUqqnZsebnu6OY0U0r924JZfztXPafF1v+18=";
+  };
+
+  build-system = [ setuptools ];
+
+  buildInputs = [ pytest ];
+
+  dependencies = [ attrs ];
+
+  optional-dependencies = {
+    docker-compose-v1 = [ docker-compose ];
+  };
+
+  #Bypass the /homeless/shelter error
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  nativeCheckInputs = [
+    py
+    pytestCheckHook
+    mypy
+    pytest-mypy
+    pytest-pycodestyle
+    pytest-pylint
+    requests
+    types-requests
+    types-setuptools
```

```suggestion
    pytestCheckHook
    requests
```
We don't need static analysis because we test to make sure it works correctly.
Please remove the pytest flag with postPatch or something.
```nix
postPatch = ''
  sed -i "/addopts =/d" setup.cfg
'';
```